docker/entrypoint.sh

```diff
@@ -3,21 +3,26 @@
 # print all commands before executing them
 set -x
 
-# create a random Namespace ID for your rollup to post blocks to
-NAMESPACE_ID=$(echo $RANDOM | md5sum | head -c 16; echo)
-echo "NAMESPACE_ID: $NAMESPACE_ID"
+# Read out the namespace ID generated in setup.sh
+# Note: If you want to regerneate the namespace ID, you need to rebuild the container
+NAMESPACE_ID=$(cat $HOME/NAMESPACE_ID)
 
 # query the DA Layer start height, in this case we are querying
 # our local devnet at port 26657, the RPC. The RPC endpoint is
 # to allow users to interact with Celestia's nodes by querying
 # the node's state and broadcasting transactions on the Celestia
 # network. The default port is 26657.
-DA_BLOCK_HEIGHT=$(curl --silent ${RPC%/}${RPC:+/}block | jq -r '.result.block.header.height')
-echo "DA_BLOCK_HEIGHT: $DA_BLOCK_HEIGHT"
+# If DA_BLOCK_HEIGHT not already set, fetch current height
+# This is needed if you want to test multiple nodes
+if [ -z "$DA_BLOCK_HEIGHT" ]; then
+    DA_BLOCK_HEIGHT=$(curl --silent ${RPC%/}${RPC:+/}block | jq -r '.result.block.header.height')
+fi
 
+# If DA_BLOCK_HEIGHT not set manually or fetched from RPC, exit
 if [ -z "$DA_BLOCK_HEIGHT" ]; then
     echo "DA_BLOCK_HEIGHT is empty; ensure that the DA Layer is running and accessible"
     exit 1
 fi
+echo "DA_BLOCK_HEIGHT: $DA_BLOCK_HEIGHT"
```

Comment:
I would add curly brackets to the variable as a best practices:

```suggestion
echo "DA_BLOCK_HEIGHT: ${DA_BLOCK_HEIGHT}"
```

Reply:
done